duckduckgo/Android
# Add handling of app links

Task/Issue URL: https://app.asana.com/0/72649045549333/1199533149091655/f
Tech Design URL: https://app.asana.com/0/481882893211075/1200311072033594
CC: 

**Description**:
This PR adds handling of App Links.

When a user taps an app link they will be prompted to open the link in the corresponding app (By tapping the "Yes" button) or view the content in the browser (By tapping the "No" button"). If there are multiple apps that can handle the link, then a chooser is shown.

The feature can be enabled/disabled in settings. It is enabled by default.

**Screenshots**:
| Dialog  | Chooser | Setting |
| ------ | ----- | ----- | 
![Screenshot_1623242574](https://user-images.githubusercontent.com/3471025/121357313-0129da80-c932-11eb-8cb2-d3fd2803f4ba.png)|![Screenshot_1623242681](https://user-images.githubusercontent.com/3471025/121357374-0edf6000-c932-11eb-9947-f01b77a88d43.png)|![Screenshot_1623242587](https://user-images.githubusercontent.com/3471025/121357474-23235d00-c932-11eb-9c99-b80e4eb74338.png)|





**Steps to test this PR**:
1. Search for "Google Maps".
2. Tap the first link.
3. Select "Yes" from the dialog and it should navigate to the Google Maps app.
4. Go back and tap the link again and select "No" from the dialog. It should open the link in the browser.
5. Go to settings, disable "Open Links in Associated Apps".
6. Tap the link again and verify that the dialog does not show and that link is opened in the browser.
7. Enable the setting again.
7. Install multiple apps that can handle a link: Twitter and Twitter Lite.
8. Search for "Twitter DuckDuckGo".
9. Tap the first link and choose "Yes" from the dialog.
10. Verify that the chooser is shown with associated apps.
11. Type in a URL associated with an app, such as maps.google.com. Verify that the link is navigated to in the browser.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
